include/utility.hpp

```diff
@@ -303,6 +305,46 @@ template <class CB_CORE, typename E> class TEnumInfo {
     }
     info.labels.elements = &clabels[0];
     info.labels.len = uint32_t(labels.size());
+
+    for (auto &v : vseq) {
+      values.emplace_back(CBEnum(v));
+    }
+    info.values.elements = &values[0];
+    info.values.len = uint32_t(values.size());
+
+    assert(info.values.len == info.labels.len);
+    CB_CORE::registerEnumType(vendorId, enumId, info);
+  }
+};
+
+template <class CB_CORE, typename E> class TFlagsInfo {
```

Comment:
Isn't this repeating the previous class but just changing some types?
Can prob template that part.

Reply:
Power of `constexpr` 😄:
```cpp
template <class CB_CORE, typename E, bool isFlag = false> class TEnumInfo {
private:
  static constexpr auto enum_names() {
    if constexpr (isFlag) {
      return magic_enum::flags::enum_names<E>();
    } else {
      return magic_enum::enum_names<E>();
    }
  }
  static constexpr auto enum_values() {
    if constexpr (isFlag) {
      return magic_enum::flags::enum_values<E>();
    } else {
      return magic_enum::enum_values<E>();
    }
  }
  static constexpr auto eseq = enum_names();
  static constexpr auto vseq = enum_values();
```
It really feels magical.